components/projects/NewProject/NewCustomer.tsx

```diff
@@ -9,28 +11,92 @@ import {
     ModalContent,
     ModalBody,
     ModalFooter,
+    Box,
 } from '@chakra-ui/react';
-import React from 'react';
+import React, { useState } from 'react';
+import { components } from '@/lib/types/api';
+import * as fetch from '@/lib/utils/fetch';
+import { FormStatus } from './reducer';
+import { useSWRConfig } from 'swr';
+import ErrorAlert from '@/components/common/ErrorAlert';
+
+type StateType = {
+    name: string;
+    description: string;
+    formStatus: FormStatus;
+};
 
 function NewCustomer(): JSX.Element {
     const { isOpen, onOpen, onClose } = useDisclosure();
+    const [state, setState] = useState<StateType>({ name: '', description: '', formStatus: FormStatus.IDLE });
+    const url = `${process.env.NEXT_PUBLIC_API_URL}/customer`;
+    const { mutate } = useSWRConfig();
+
+    const handleSubmit = async (e: React.MouseEvent) => {
+        e.preventDefault();
+
+        const createCustomerRequest: components['schemas']['CustomerDTO'] = {
+            name: state.name,
+            description: state.description,
+        };
+        setState({ name: state.name, description: state.description, formStatus: FormStatus.LOADING });
+        try {
+            await fetch.post(url, createCustomerRequest);
+            mutate(url);
+            setState({ name: state.name, description: state.description, formStatus: FormStatus.SUCCESS });
+            onClose();
+        } catch (error) {
+            setState({ name: state.name, description: state.description, formStatus: FormStatus.ERROR });
+        }
```

Comment:
This error should probably be added to some notification? Or are we just catching errors without trying to find out what the error is?

Reply:
One way would be to have `errorMessage: string` or similar as a state parameter, which is displayed if it has text

Reply:
Added errorMessage to state in 9137767. Error message is shown in a Box element. I think it should be a part of ErrorAlert component, but that will have to wait for tomorrow